### PR TITLE
[posix] ignore unexpected host netif state notification

### PR DESCRIPTION
**Pull request by bukepo**

### CODE REVIEW

File: src/posix/platform/netif.cpp [hunk 1]
```diff
@@ -768,18 +775,32 @@ static void processNetifLinkEvent(otInstance *aInstance, struct nlmsghdr *aNetli
...
-    if (error == OT_ERROR_NONE)
+    isUp = ((ifinfo->ifi_flags & IFF_UP) != 0);
+
+    otLogInfoPlat("Host netif is %s", isUp ? "up" : "down");
+
+#if defined(RTM_NEWLINK) && defined(RTM_DELLINK)
+    if (sIsSyncingState)
     {
-        otLogInfoPlat("%s: %s", __func__, otThreadErrorToString(error));
+        VerifyOrExit(isUp == otIp6IsEnabled(aInstance), otLogWarnPlat("Host netif state notfication is unexpected"));
```
Comment (simonlingoogle) [inline src/posix/platform/netif.cpp:789]:
```suggestion
        VerifyOrExit(isUp == otIp6IsEnabled(aInstance), otLogWarnPlat("Host netif state notification is unexpected (ignored)"));
```

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: with nits

This PR allows OpenThread master to work with ot-br-posix, so that we can update OT commit to master (https://github.com/openthread/ot-br-posix/pull/594)

**Review by jwhui — APPROVED** [review]

File: src/posix/platform/netif.cpp [hunk 2]
```diff
@@ -768,18 +775,32 @@ static void processNetifLinkEvent(otInstance *aInstance, struct nlmsghdr *aNetli
...
-    if (error == OT_ERROR_NONE)
+    isUp = ((ifinfo->ifi_flags & IFF_UP) != 0);
+
+    otLogInfoPlat("Host netif is %s", isUp ? "up" : "down");
+
+#if defined(RTM_NEWLINK) && defined(RTM_DELLINK)
+    if (sIsSyncingState)
     {
-        otLogInfoPlat("%s: %s", __func__, otThreadErrorToString(error));
+        VerifyOrExit(isUp == otIp6IsEnabled(aInstance), otLogWarnPlat("Host netif state notification is unexpected (ignored)"));
```
Comment (jwhui) [inline src/posix/platform/netif.cpp:789]: nit:
```suggestion
        VerifyOrExit(isUp == otIp6IsEnabled(aInstance), otLogWarnPlat("Host netif state notification is unexpected (ignore)"));
```

**Comment by jwhui** [comment]
@bukepo , pretty check is failing.

File: src/posix/platform/netif.cpp [hunk 3]
```diff
@@ -768,18 +775,33 @@ static void processNetifLinkEvent(otInstance *aInstance, struct nlmsghdr *aNetli
...
+    isUp = ((ifinfo->ifi_flags & IFF_UP) != 0);
+
+    otLogInfoPlat("Host netif is %s", isUp ? "up" : "down");
+
+#if defined(RTM_NEWLINK) && defined(RTM_DELLINK)
+    if (sIsSyncingState)
     {
-        otLogInfoPlat("%s: %s", __func__, otThreadErrorToString(error));
+        VerifyOrExit(isUp == otIp6IsEnabled(aInstance),
+                     otLogWarnPlat("Host netif state notification is unexpected (ignored)"));
```
Comment (jwhui) [inline src/posix/platform/netif.cpp:790]:
```suggestion
                     otLogWarnPlat("Host netif state notification is unexpected (ignore)"));
```

**Pull request merged**